contrib/slurm/slurmexecutor.py

```diff
@@ -0,0 +1,297 @@
+    benchmark.executable = benchmark.tool.executable(tool_locator)
+    benchmark.tool_version = benchmark.tool.version(benchmark.executable)
+
+    logging.info("Using %s version %s.", benchmark.tool_name, benchmark.tool_version)
+
+
+def get_system_info():
+    return None
+
+
+def execute_benchmark(benchmark, output_handler):
+    num_of_cores = benchmark.rlimits.cpu_cores
+    mem_limit = benchmark.rlimits.memory
+    run_sets_executed = 0
+
+    for runSet in benchmark.run_sets:
+        if STOPPED_BY_INTERRUPT:
+            break
+
+        if not runSet.should_be_executed():
+            output_handler.output_for_skipping_run_set(runSet)
+
+        elif not runSet.runs:
+            output_handler.output_for_skipping_run_set(
+                runSet, "because it has no files"
+            )
+
+        else:
+            run_sets_executed += 1
+            _execute_run_set(
+                runSet,
+                benchmark,
+                output_handler,
+                num_of_cores,
+                mem_limit,
+            )
+
+    output_handler.output_after_benchmark(STOPPED_BY_INTERRUPT)
+
+
+def _execute_run_set(
+    runSet,
+    benchmark,
+    output_handler,
+    num_of_cores,
+    mem_limit,
+):
+    # get times before runSet
+    walltime_before = time.monotonic()
+
+    output_handler.output_before_run_set(runSet)
+
+    # put all runs into a queue
+    for run in runSet.runs:
+        _Worker.working_queue.put(run)
+
+    # keep a counter of unfinished runs for the below assertion
+    unfinished_runs = len(runSet.runs)
+    unfinished_runs_lock = threading.Lock()
+
+    def run_finished():
+        nonlocal unfinished_runs
+        with unfinished_runs_lock:
+            unfinished_runs -= 1
+
+    # create some workers
+    for i in range(min(benchmark.num_of_threads, unfinished_runs)):
+        if STOPPED_BY_INTERRUPT:
+            break
+        WORKER_THREADS.append(
+            _Worker(benchmark, num_of_cores, mem_limit, output_handler, run_finished)
+        )
+
+    # wait until workers are finished (all tasks done or STOPPED_BY_INTERRUPT)
+    for worker in WORKER_THREADS:
+        worker.join()
+    assert unfinished_runs == 0 or STOPPED_BY_INTERRUPT
+
+    # get times after runSet
+    walltime_after = time.monotonic()
+    usedWallTime = walltime_after - walltime_before
+    usedCpuTime = 1000  # TODO
+
+    if STOPPED_BY_INTERRUPT:
+        output_handler.set_error("interrupted", runSet)
+    output_handler.output_after_run_set(
+        runSet,
+        cputime=usedCpuTime,
+        walltime=usedWallTime,
+    )
+
+
+def stop():
+    global STOPPED_BY_INTERRUPT
+    STOPPED_BY_INTERRUPT = True
+
+
+class _Worker(threading.Thread):
+    """
+    A Worker is a deamonic thread, that takes jobs from the working_queue and runs them.
+    """
+
+    working_queue = queue.Queue()
+
+    def __init__(
+        self, benchmark, my_cpus, my_memory_nodes, output_handler, run_finished_callback
```

Comment:
`my_cpus` is a misleading name, because it seems like it is a list of CPU cores. Please rename. And please remove unused parameters (`my_memory_nodes`).

Reply:
Done in the rework of params.